src/XMakeBuildEngine/Utilities/Utilities.cs

```diff
@@ -459,7 +459,7 @@ internal static PropertyDictionary<ProjectPropertyInstance> GetEnvironmentProper
             // environment or as a global property.  
 
 #if !FEATURE_INSTALLED_MSBUILD
```

Comment:
Strong disagree on that. The advantage of flagging for individual features is that it provides information about why things are changed in different flavors. Consider this example: it now doesn't really represent its name, because what it really means is `FEATURE_PROGRAMFILES_EXTENSION_ROOT` or something. But since we now know that `INSTALLED_MSBUILD` is a bad name, we can look at all uses of it and figure out what to do now. If it was just behind `!FULL`, there would be many more cases to look at.

Or think about `FEATURE_REGISTRY`--we can decide to enable that for windows, which is just finding the ifdevs and replacing them with runtime checks--but only the ones with a given label.